### PR TITLE
Check your answers

### PR DESCRIPTION
**Pull request by DavidJayakumar**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/EUI-8724

### Change description ###

Add check your answers component

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/register-org/components/check-your-answers/check-your-answers.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { HttpClientTestingModule } from '@angular/common/http/testing';
```
Comment (connorpgpmcelroy) [inline src/register-org/components/check-your-answers/check-your-answers.component.spec.ts:3]: minor alphabetic ordering issue

**Review by connorpgpmcelroy — APPROVED** [review]
LGTM, apart from minor comment that is not necessary to fix

File: src/register-org/components/check-your-answers/check-your-answers.component.html [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+              [routerLink]="'/register-org-new/registered-with-regulator'">
+              Change<span class="govuk-visually-hidden">Regulatory organisation type</span>
+            </a>
+          </dd>
+        </div>
+      </dl>
+
+      <dl class="govuk-summary-list govuk-!-margin-bottom-0">
+        <div class="govuk-summary-list__row">
+          <dt class="govuk-summary-list__key">What regulator are you(as an individual) registered with?</dt>
```
Comment (Daniel-Lam) [inline src/register-org/components/check-your-answers/check-your-answers.component.html:101]: @DavidJayakumar - needs a space between `you` and `(as`

File: src/register-org/components/check-your-answers/check-your-answers.component.html [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+          <dd class="govuk-summary-list__value">{{ registrationData.reulatorRegisteredWith }}</dd>
+          <dd class="govuk-summary-list__actions">
+            <a class="govuk-link" href="javascript:void(0)"
+              [routerLink]="'/register-org-new/registered-with-regulator'">
+              Change<span class="govuk-visually-hidden">What regulator are you(as an individual) registered with?</span>
```
Comment (Daniel-Lam) [inline src/register-org/components/check-your-answers/check-your-answers.component.html:106]: @DavidJayakumar - needs a space between `you` and `(as`

File: src/register-org/components/check-your-answers/check-your-answers.component.html [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+        </div>
+      </dl>
+      <br>
+      <h2 class="govuk-heading-m">Confirm your registration details</h2>
+      <p>Ensure that:</p>
+      <ul class="govuk-list govuk-list--bullet">
+        <li>your personal and organisation detils are correct</li>
+        <li>you have used your registered office address and not a personal or home address</li>
+        <li>you are creating an account for your organisation for the first time</li>
+        <li>you are authorised to register an account on behalf of your ogrnaisation</li>
```
Comment (Daniel-Lam) [inline src/register-org/components/check-your-answers/check-your-answers.component.html:156]: @DavidJayakumar - spelling mistake, should be `organisation`

File: src/register-org/components/check-your-answers/check-your-answers.component.html [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+      <h1 class="govuk-heading-l">Check your answers before you register</h1>
+      <h2 class="govuk-heading-m">Organisation details</h2>
+
+      <dl class="govuk-summary-list govuk-!-margin-bottom-0">
+        <div class="govuk-summary-list__row">
+          <dt class="govuk-summary-list__key">Organisation name</dt>
+          <dd class="govuk-summary-list__value">{{ registrationData.name }}</dd>
+          <dd class="govuk-summary-list__actions">
+            <a class="govuk-link" href="javascript:void(0)"
+              [routerLink]="'/register-org-new/company-house-details'">Change
```
Comment (johnbenjamin-hmcts) [inline src/register-org/components/check-your-answers/check-your-answers.component.html:14]: @DavidJayakumar - please use this property `REGISTER_ORG_NEW_ROUTE` from `RegisterOrgService` as suggested by Ronald for one of my older PR.

File: src/register-org/components/check-your-answers/check-your-answers.component.html [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+        </div>
+      </dl>
+
+      <dl *ngIf="registrationData.companyHouseNumber" class="govuk-summary-list govuk-!-margin-bottom-0">
+        <div class="govuk-summary-list__row">
+          <dt class="govuk-summary-list__key">Company registration number</dt>
+          <dd class="govuk-summary-list__value">{{ registrationData.companyHouseNumber }}</dd>
+          <dd class="govuk-summary-list__actions">
+            <a class="govuk-link" href="javascript:history.back()"
+              [routerLink]="'/register-org-new/company-house-details'">
```
Comment (johnbenjamin-hmcts) [inline src/register-org/components/check-your-answers/check-your-answers.component.html:27]: @DavidJayakumar - please use this property `REGISTER_ORG_NEW_ROUTE` from `RegisterOrgService` as suggested by Ronald for one of my older PR.

File: src/register-org/components/check-your-answers/check-your-answers.component.html [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+        </div>
+      </dl>
+
+      <dl *ngIf="registrationData.dxNumber" class="govuk-summary-list govuk-!-margin-bottom-0">
+        <div class="govuk-summary-list__row">
+          <dt class="govuk-summary-list__key">DX reference</dt>
+          <dd class="govuk-summary-list__value">{{ registrationData.dxNumber }}</dd>
+          <dd class="govuk-summary-list__actions">
+            <a class="govuk-link" href="javascript:void(0)"
+              [routerLink]="'/register-org-new/document-exchange-reference-details'">
```
Comment (johnbenjamin-hmcts) [inline src/register-org/components/check-your-answers/check-your-answers.component.html:53]: @DavidJayakumar - please use this property `REGISTER_ORG_NEW_ROUTE` from `RegisterOrgService` as suggested by Ronald for one of my older PR.

File: src/register-org/components/check-your-answers/check-your-answers.component.html [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+      </dl>
+
+
+      <dl class="govuk-summary-list govuk-!-margin-bottom-0 govuk-summary-list--no-border">
+        <div class="govuk-summary-list__row">
+          <dt class="govuk-summary-list__key">Service to access</dt>
+          <dd class="govuk-summary-list__value">{{ registrationData.services }}</dd>
+          <dd class="govuk-summary-list__actions">
+            <a class="govuk-link" href="javascript:void(0)"
+              [routerLink]="'/register-org-new/organisation-services-access'">
```
Comment (johnbenjamin-hmcts) [inline src/register-org/components/check-your-answers/check-your-answers.component.html:67]: @DavidJayakumar - same as above.

File: src/register-org/components/check-your-answers/check-your-answers.component.html [hunk 9]
```diff
@@ -0,0 +1,194 @@
...
+        </div>
+      </dl>
+
+      <dl class="govuk-summary-list govuk-!-margin-bottom-0 govuk-summary-list--no-border">
+        <div class="govuk-summary-list__row">
+          <dt class="govuk-summary-list__key">Regulatory organisation type</dt>
+          <dd class="govuk-summary-list__value">{{ registrationData.regulatoryOrgType }}</dd>
+          <dd class="govuk-summary-list__actions">
+            <a class="govuk-link" href="javascript:void(0)"
+              [routerLink]="'/register-org-new/registered-with-regulator'">
```
Comment (johnbenjamin-hmcts) [inline src/register-org/components/check-your-answers/check-your-answers.component.html:92]: @DavidJayakumar - same as above.

File: src/register-org/components/check-your-answers/check-your-answers.component.html [hunk 10]
```diff
@@ -0,0 +1,194 @@
...
+        </div>
+      </dl>
+
+      <dl class="govuk-summary-list govuk-!-margin-bottom-0">
+        <div class="govuk-summary-list__row">
+          <dt class="govuk-summary-list__key">What regulator are you(as an individual) registered with?</dt>
+          <dd class="govuk-summary-list__value">{{ registrationData.reulatorRegisteredWith }}</dd>
+          <dd class="govuk-summary-list__actions">
+            <a class="govuk-link" href="javascript:void(0)"
+              [routerLink]="'/register-org-new/registered-with-regulator'">
```
Comment (johnbenjamin-hmcts) [inline src/register-org/components/check-your-answers/check-your-answers.component.html:105]: @DavidJayakumar - same as above.

File: src/register-org/components/check-your-answers/check-your-answers.component.html [hunk 11]
```diff
@@ -0,0 +1,194 @@
...
+          </dd>
+        </div>
+      </dl>
+
+      <dl class="govuk-summary-list govuk-!-margin-bottom-0">
+        <div class="govuk-summary-list__row">
+          <dt class="govuk-summary-list__key">Organisation type</dt>
+          <dd class="govuk-summary-list__value">{{ registrationData.organisationType }}</dd>
+          <dd class="govuk-summary-list__actions">
+            <a class="govuk-link" href="javascript:void(0)" [routerLink]="'/register-org-new/organisation-type'">
```
Comment (johnbenjamin-hmcts) [inline src/register-org/components/check-your-answers/check-your-answers.component.html:79]: @DavidJayakumar - same as above.

File: src/register-org/components/check-your-answers/check-your-answers.component.ts [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+import { RegisterComponent } from '../../../register-org/containers';
+import { RegisterOrgService } from '../../services/register-org.service';
+
+@Component({
+  selector: 'app-check-your-answers',
+  templateUrl: './check-your-answers.component.html'
+})
+export class CheckYourAnswersComponent extends RegisterComponent implements OnInit {
+  constructor(public readonly router: Router,
+    public readonly registerOrgService: RegisterOrgService,
```
Comment (Daniel-Lam) [inline src/register-org/components/check-your-answers/check-your-answers.component.ts:12]: @DavidJayakumar - does this injection need to be `public`?

File: src/register-org/components/check-your-answers/check-your-answers.component.html [hunk 13]
```diff
@@ -0,0 +1,194 @@
...
+      </dl>
+      <br>
+      <h2 class="govuk-heading-m">Administration details</h2>
+
+      <dl class="govuk-summary-list govuk-!-margin-bottom-0">
+        <div class="govuk-summary-list__row">
+          <dt class="govuk-summary-list__key">First name</dt>
+          <dd class="govuk-summary-list__value">{{ registrationData.contactDetails.firstName }}</dd>
+          <dd class="govuk-summary-list__actions">
+            <a class="govuk-link" href="javascript:void(0)" [routerLink]="'/register-org-new/contact-details'">
```
Comment (johnbenjamin-hmcts) [inline src/register-org/components/check-your-answers/check-your-answers.component.html:119]: @DavidJayakumar - same as above.

File: src/register-org/components/check-your-answers/check-your-answers.component.html [hunk 14]
```diff
@@ -0,0 +1,194 @@
...
+          </dd>
+        </div>
+      </dl>
+
+      <dl class="govuk-summary-list govuk-!-margin-bottom-0">
+        <div class="govuk-summary-list__row">
+          <dt class="govuk-summary-list__key">Last name</dt>
+          <dd class="govuk-summary-list__value">{{ registrationData.contactDetails.lastName }}</dd>
+          <dd class="govuk-summary-list__actions">
+            <a class="govuk-link" href="javascript:void(0)" [routerLink]="'/register-org-new/contact-details'">
```
Comment (johnbenjamin-hmcts) [inline src/register-org/components/check-your-answers/check-your-answers.component.html:131]: @DavidJayakumar - same as above.

File: src/register-org/components/check-your-answers/check-your-answers.component.html [hunk 15]
```diff
@@ -0,0 +1,194 @@
...
+          </dd>
+        </div>
+      </dl>
+
+      <dl class="govuk-summary-list govuk-!-margin-bottom-0">
+        <div class="govuk-summary-list__row">
+          <dt class="govuk-summary-list__key">Email address</dt>
+          <dd class="govuk-summary-list__value">{{ registrationData.contactDetails.workEmailAddress }}</dd>
+          <dd class="govuk-summary-list__actions">
+            <a class="govuk-link" href="javascript:void(0)" [routerLink]="'/register-org-new/contact-details'">
```
Comment (johnbenjamin-hmcts) [inline src/register-org/components/check-your-answers/check-your-answers.component.html:143]: @DavidJayakumar - same as above.

File: src/register-org/models/registrationdata.model.ts [hunk 16]
```diff
@@ -9,4 +9,10 @@ export type RegistrationData = {
   hasPBA: boolean;
   contactDetails: ContactDetails;
   hasRegisteredWithRegulator: boolean;
+  companyHouseNumber?: string;
+  address: string;
+  organisationType: string;
+  organisatinonNumber?: string;
```
Comment (Daniel-Lam) [inline src/register-org/models/registrationdata.model.ts:15]: @DavidJayakumar - looks like a typo here; perhaps you meant `organisationNumber`?

File: src/register-org/models/registrationdata.model.ts [hunk 17]
```diff
@@ -9,4 +9,10 @@ export type RegistrationData = {
...
+  regulatoryOrgType: string;
+  reulatorRegisteredWith: string;
```
Comment (Daniel-Lam) [inline src/register-org/models/registrationdata.model.ts:17]: @DavidJayakumar - looks like a typo here; perhaps you meant `regulatorRegisteredWith`?

File: src/register-org/components/registered-with-regulator/registered-with-regulator.component.spec.ts [hunk 18]
```diff
@@ -16,7 +16,11 @@ describe('RegisteredWithRegulatorComponent', () => {
     hasPBA: null,
     contactDetails: null,
     hasRegisteredWithRegulator: null,
-    services: []
+    services: [],
+    address: null,
+    organisationType: null,
+    regulatoryOrgType: null,
+    reulatorRegisteredWith: null
```
Comment (Daniel-Lam) [inline src/register-org/components/registered-with-regulator/registered-with-regulator.component.spec.ts:23]: @DavidJayakumar - looks like a typo here; perhaps you meant `regulatorRegisteredWith`?

**Review by johnbenjamin-hmcts — CHANGES_REQUESTED** [review]

**Review by Daniel-Lam — CHANGES_REQUESTED** [review]
@DavidJayakumar - some changes needed. Also, please remove the empty `company-house-details.component.scss` file and its reference in the corresponding component

File: src/register-org/components/check-your-answers/check-your-answers.component.html [hunk 19]
```diff
@@ -0,0 +1,194 @@
...
+        <li>you have used your registered office address and not a personal or home address</li>
+        <li>you are creating an account for your organisation for the first time</li>
+        <li>you are authorised to register an account on behalf of your ogrnaisation</li>
+      </ul>
+
+      <div class="govuk-warning-text">
+        <span class="govuk-warning-text__icon" aria-hidden="true">!</span>
+        <strong class="govuk-warning-text__text">
+          <span class="govuk-warning-text__assistive">Warning</span>
+          Make sure all information is correct as your appliction will be rejected if it contains errors.
```
Comment (Daniel-Lam) [inline src/register-org/components/check-your-answers/check-your-answers.component.html:163]: @DavidJayakumar - typo here; should be `application`

File: src/register-org/components/payment-by-account/payment-by-account.component.spec.ts [hunk 20]
```diff
@@ -18,7 +18,11 @@ describe('PaymentByAccountComponent', () => {
     services: [],
     hasPBA: null,
     contactDetails: null,
-    hasRegisteredWithRegulator: null
+    hasRegisteredWithRegulator: null,
+    address: null,
+    organisationType: null,
+    regulatoryOrgType: null,
+    reulatorRegisteredWith: null
```
Comment (Daniel-Lam) [inline src/register-org/components/payment-by-account/payment-by-account.component.spec.ts:25]: @DavidJayakumar - looks like a typo here; perhaps you meant `regulatorRegisteredWith`?

File: src/register-org/components/check-your-answers/check-your-answers.component.html (hunk 19 above)
Comment (DavidJayakumar) [reply src/register-org/components/check-your-answers/check-your-answers.component.html]: done

File: src/register-org/components/check-your-answers/check-your-answers.component.html (hunk 5 above)
Comment (DavidJayakumar) [reply src/register-org/components/check-your-answers/check-your-answers.component.html]: Fixed all

**Review by Daniel-Lam — APPROVED** [review]

**Review by johnbenjamin-hmcts — APPROVED** [review]

**Pull request merged**